manifests/config/passenger.pp

```diff
@@ -175,6 +179,67 @@
         custom_fragment         => template('foreman/_assets.conf.erb', 'foreman/_ssl_virt_host_include.erb',
                                             'foreman/_suburi.conf.erb'),
       }
+
+      $bundle_cert_dir = '/usr/share/foreman/ssl'
+      $client_bundle = "${bundle_cert_dir}/client_bundle.pem"
+
+      file { $bundle_cert_dir:
+        ensure => 'directory',
+        owner  => 'root',
+        group  => $::foreman::group,
+        mode   => '0750',
+      }
+
+      concat {$client_bundle:
+        owner => 'root',
+        group => $::foreman::group,
+        mode  => '0640',
+      }
+
+      concat::fragment{ 'client_key':
+        target => $client_bundle,
+        source => $client_key,
+        order  => '01',
+      }
+
+      concat::fragment{ 'client_cert':
+        target => $client_bundle,
+        source => $client_cert,
+        order  => '02',
+      }
+
+      $infrastructure_url = "https://${servername}:${infrastructure_port}"
+      $infrastructure_redirection_path = '/'
+      apache::vhost { 'foreman-ssl':
+        add_default_charset => 'UTF-8',
+        docroot             => $docroot,
+        ip                  => $listen_interface,
+        options             => ['SymLinksIfOwnerMatch'],
+        port                => 443,
+        priority            => '05',
+        servername          => $servername,
+        serveraliases       => ['foreman'],
+        ssl                 => true,
+        ssl_cert            => $ssl_cert,
+        ssl_key             => $ssl_key,
+        ssl_chain           => $ssl_chain,
+        ssl_ca              => $ssl_ca,
+        ssl_crl             => $ssl_crl_real,
+        ssl_crl_check       => $ssl_crl_check,
+        ssl_verify_client   => 'optional',
+        ssl_options         => '+StdEnvVars +ExportCertData',
+        ssl_verify_depth    => '3',
+        ssl_proxyengine     => true,
+        proxy_pass          => [{
+          'path'         => $infrastructure_redirection_path,
+          'url'          => $infrastructure_url,
+          'reverse_urls' => [$infrastructure_redirection_path, $infrastructure_url]
+        }],
+        custom_fragment     => "
+        SSLProxyCACertificateFile ${client_ca}
+        SSLProxyMachineCertificateFile ${client_bundle}
```

Comment:
After some testing using (https://github.com/iNecas/ownca) to generate custom certificates for 443 port, I agree with @ekohl that this is not needed as I was able to do a full install which included a smart proxy registering to 443 successfully.